applications/StructuralMechanicsApplication/tests/test_adjoint_sensitity_analysis_shell_3d3n_structure.py

```diff
@@ -39,6 +39,16 @@ def solve_primal_problem():
     primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(model_primal, ProjectParametersPrimal)
     primal_analysis.Run()
 
+def run_solution_loop_adjoint_senstivity_analysis(adjoint_analysis, adjoint_postprocess):
+    while adjoint_analysis.time < adjoint_analysis.end_time:
+        adjoint_analysis.time = adjoint_analysis._GetSolver().AdvanceInTime(adjoint_analysis.time)
+        adjoint_analysis.InitializeSolutionStep()
+        adjoint_analysis._GetSolver().Predict()
+        adjoint_analysis._GetSolver().SolveSolutionStep()
+        adjoint_postprocess.UpdateSensitivities()
```

Comment:
I see your point. But there is currently no AdjointAnalysis and performing UpdateSensitivities should be independent from solving the adjoint problem. The only reason why I compute the sensitivities here within the interface of the analysis stage is that I like to use the ```from_json_check_result_process```. So I need the sensitivities before FinalizeSolutionStep. @philbucher you think we should have an own custom AdjointAnalysis instead of having some python running scripts were we have to extend the interface of the AnalysisStage?

Reply:
ok I understand. For now it is ok.

You could think if an AdjointAnalysis would make sense (in Fluid they have it)
Maybe this would also make it easier to use it from the ShapeOpt (@dbaumgaertner @armingeiser ??)

Then you can easily do what you want to do without extending the interface

Reply:
In ShapeOptimization they use the ```structural_response.py``` which is very similar to a custom analysis. Maybe I will establish something similar for my purposes.

Reply:
I would love to have the adjoint_postprocess as a real `process`, but it needs the response_function object (currently passed in the constructor), therefor it cannot be created in the list_of_processes of the base analysis.
If we find a smart way to pass the response object somehow to the adjoint_postprocess, we could even use the base analysis...

I don't know if storing a pointer to the response at the modelpart somehow is a smart idea?

Reply:
you could have a custom analysis stage that constructs the adjoint-postprocess as a real process in `_CreateProcesses` => then you can pass the response in the constructor but it will still be used as a regular process

I can explain more in person if you want